typescript-json-schema.ts

```diff
@@ -607,9 +604,10 @@ export class JsonSchemaGenerator {
                 pushEnumValue(value);
             } else {
                 const def = this.getTypeDefinition(valueType);
-                if (def.type === "undefined") {
+                if ((<any>def).mayBeUndefined) {
```

Comment:
Can't say I'm a fan of this. 

If I understand correctly, we only want to make sure that we update the `required` property correctly, right? SO I think it really only matters in https://github.com/YousefED/typescript-json-schema/blob/921dc104f5fc3231a3e32b34585f05a04e08480a/typescript-json-schema.ts#L816. 

As a proposal, can you assemble the required properties you go through them in `getClassDefinition` and then in `getUnionDefinition` just look at the `required` property? This would avoid using `any`.